src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -160,12 +160,17 @@ namespace System.Diagnostics
     /// See the DiagnosticSourceEventSourceBridgeTest.cs for more explicit examples of using this bridge.
     /// </summary>
     [EventSource(Name = "Microsoft-Diagnostics-DiagnosticSource")]
-    // This coarse suppression silences all RequiresUnreferencedCode warnings in the class.
-    // https://github.com/mono/linker/issues/2136 tracks making it possible to add more granular suppressions at the member level, and with a different warning code.
-    [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
-        Justification = "In EventSource, EnsureDescriptorsInitialized's use of GetType preserves all members, so " +
-                        "those that are marked with RequiresUnreferencedCode will warn. " +
-                        "This method will not access any of these members and is safe to call.")]
+    // These suppressions can go away with https://github.com/mono/linker/issues/2175
+    [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2113:RequiresUnreferencedCode",
+        Justification = "In EventSource, EnsureDescriptorsInitialized's use of GetType preserves methods on Delegate and MulticastDelegate " +
+                        "because the nested type OverrideEventProvider's base type EventProvider defines a delegate. " +
```

Comment:
I'm not sure I understand why `DiagnosticSourceEventSource` is getting warnings for `OverrideEventProvider`. If it is - why isn't all EventSource's getting this warning?

Reply:
The warning is due to DAMT.All on EventSource, which ensures that GetType keeps _everything_ on EventSource - in this case the nested type OverrideEventProvider, plus its base type EventProvider, which has a [delegate](https://github.com/dotnet/runtime/blob/f9cbd72baba7f76f32b355fbc3da735f5d8bacfb/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventProvider.cs#L450), which has some RUC methods: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Delegate.cs#L45.

Ideally we wouldn't produce this warning for derived types, and it would show up only on EventSource, but we haven't taught the linker about this (see https://github.com/mono/linker/issues/2175).

In theory this should currently show up on every derived EventSource... but I believe due to https://github.com/mono/linker/issues/2159 we only produce a warning for the first one that we see. :( /cc @LakshanF

Reply:
> we only produce a warning for the first one that we see. :( 

Oh no, that doesn't sound like a great situation.

Reply:
This sounds bad - could it lead to us getting warnings from CoreLib when trimming apps? For some reason linker will see this on another EventSource and will warn there - and it's not suppressed there. We need to fix this if that's the case.